docs/develop/configure-your-add-in-to-use-a-shared-runtime.md

```diff
@@ -0,0 +1,221 @@
+
+>[!NOTE]
+> The default lifetime value is `short`, but we recommend using `long` in Excel add-ins. If you set your runtime to `short` in this example, your Excel add-in will start when one of your ribbon buttons is pressed, but it may shut down after your ribbon handler is done running. Similarly your add-in will start when the task pane is opened, but it may shut down when the task pane is closed.
+
+```xml
+<Runtimes>
+  <Runtime resid="ContosoAddin.Url" lifetime="long" />
+</Runtimes>
+```
+
+>[!NOTE]
+> If your add-in includes the `Runtimes` element in the manifest (required for a shared runtime), it uses Internet Explorer 11 regardless of the Windows or Microsoft 365 version. For more information, see [Runtimes](../reference/manifest/runtimes.md).
+
+## About the shared JavaScript runtime
+
+On Windows or Mac, your add-in will run code for ribbon buttons, custom functions, and the task pane in separate JavaScript runtime environments. This creates limitations such as not being able to easily share global data, and not being able to access all CORS functionality from a custom function.
+
+However, you can configure your Office Add-in to share code in the same JavaScript runtime (also referred to as a shared runtime). This enables better coordination across your add-in and access to the task pane DOM and CORS from all parts of your add-in.
+
+Configuring a shared runtime enables the following scenarios.
+
+- Your Office Add-in can use additional UI features:
+    - [Add Custom keyboard shortcuts to your Office Add-ins (preview)](../design/keyboard-shortcuts.md)
+    - [Create custom contextual tabs in Office Add-ins (preview)](../design/contextual-tabs.md)
+    - [Enable and Disable Add-in Commands](../design/disable-add-in-commands.md)
+    - [Run code in your Office Add-in when the document opens](run-code-on-document-open.md)
+    - [Show or hide the task pane of your Office Add-in](show-hide-add-in.md)
+- For Excel add-ins:
+    - Custom functions will have full CORS support.
+    - Custom functions can call Office.js APIs to read spreadsheet document data.
+
+For Office on Windows, the shared runtime requires a Microsoft Internet Explorer 11 browser instance, as explained in [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md). Additionally, any buttons that your add-in displays on the ribbon will run in the same shared runtime. The following image shows how custom functions, the ribbon UI, and the task pane code will all run in the same JavaScript runtime.
+
+![Diagram of a custom function, task pane, and ribbon buttons all running in a shared IE/Edge browser runtime in Excel](../images/custom-functions-in-browser-runtime.png)
```

Comment:
Should the alt tag mention Edge browser?

Reply:
yep, fixed.